### ISSUE TITLE
H5::DataSetIException when using IODataset

### ISSUE THREAD

**Issue by gruossomonica** [tensorflow/io#791]
Hi, 
similar to [#745](https://github.com/tensorflow/io/issues/745), I tried to train my neural network with a dataset consisting of some hdf5 files and I got the following error:

`1511/1607 [===========================>..] - ETA: 6s - loss: 0.3598 - my_upsample_heatmap_before_proj_loss: 0.0019 - joints3D_prediction_before_proj_loss: 0.1732 - my_upsample_heatmap_before_proj_accuracy: 0.9798 - joints3D_prediction_before_proj_accuracy: 0.3065HDF5-DIAG: Error detected in HDF5 (1.10.5) thread 0:
  #000: external/hdf5/src/H5Dio.c line 199 in H5Dread(): can't read data
    major: Dataset
    minor: Read failed
  #001: external/hdf5/src/H5Dio.c line 493 in H5D__read(): no output buffer
    major: Invalid arguments to routine
    minor: Bad value
1513/1607 [===========================>..] - ETA: 5s - loss: 0.3598 - my_upsample_heatmap_before_proj_loss: 0.0019 - joints3D_prediction_before_proj_loss: 0.1732 - my_upsample_heatmap_before_proj_accuracy: 0.9798 - joints3D_prediction_before_proj_accuracy1515/1607 [===========================>..] - ETA: 5s - loss: 0.3599 - my_upsample_heatmap_before_proj_loss: 0.0019 - joints3D_prediction_before_proj_loss: 0.1732 - my_upsample_heatmap_before_proj_accuracy: 0.9798 - joints3D_prediction_before_proj_accuracy1517/1607 [===========================>..] - ETA: 5s - loss: 0.3599 - my_upsample_heatmap_before_proj_loss: 0.0019 - joints3D_prediction_before_proj_loss: 0.1732 - my_upsample_heatmap_before_proj_accuracy: 0.9798 - joints3D_prediction_before_proj_accuracy: 0.3056terminate called after throwing an instance of 'H5::DataSetIException'
Aborted (core dumped)
`

When I tried with a small hdf5 file there were no problems. No problem even when the training has been done for a few epochs. Now it has stuck at 260 epochs. I don't know which file is causing this.
My configuration:

- Ubuntu 18.04.4 LTS with kernel 4.15.0-041500-generic
- Python 3.6.9
- TensorFlow GPU 2.0.0
- TensorFlowIO 0.10.0
- Nvidia Titan Xp GPU
- CUDA 10.0 and CuDNN v7.6.5

This is my code for creating the training dataset:
```python
def load_hdf5_train_files():
	
	train_path = [os.path.join(root, name) 
	for root, dirs, files in os.walk(args.input_dir) 
	for name in files 
	if name.endswith('.hdf5') and "train" in name]

	images_train = tfio.IODataset.from_hdf5(train_path[0], '/data')
	label_train = tfio.IODataset.from_hdf5(train_path[0], '/label')
	label2_train = tfio.IODataset.from_hdf5(train_path[0], '/label2')
	print("hdf5 data: ", train_path[0])

	k = 1
	for file in train_path:
		if file != train_path[0] and k <= 50:  # 51417 images in train dataset
			k = k+1
			print("concatenating hdf5 data: ", file)
			img = tfio.IODataset.from_hdf5(file, '/data')
			#print("number of img files: ", len(list(img)))
			label = tfio.IODataset.from_hdf5(file, '/label')
			label2 = tfio.IODataset.from_hdf5(file, '/label2')
			images_train = images_train.concatenate(img)
			label_train = label_train.concatenate(label)
			label2_train = label2_train.concatenate(label2)
			print("number of images: ", len(list(images_train)))
		elif k > 50:
			break

	ds = tf.data.Dataset.zip((images_train, label_train, label2_train))

	print("number of images in train dataset: ", len(list(images_train)))

	return ds

def createTrainDataset():
	ds = load_hdf5_train_files()

	# preprocessing
	ds = ds.map(lambda data, label, label2: 
		({'input': tf.transpose(tf.image.convert_image_dtype(data,tf.float32), perm=[2, 1, 0])}, 
		{'joints3D_prediction_before_proj': tf.cast(label,tf.float32), 
		'my_upsample_heatmap_before_proj': tf.transpose(tf.cast(label2,tf.float32), perm=[2, 1, 0])}),
		num_parallel_calls=AUTOTUNE)
	print(ds)
	
	# shuffle
	#ds = ds.shuffle(buffer_size=args.shuffle_buffer_size)

	# batch
	ds = ds.batch(args.batch_size)

	# prefetch
	ds = ds.prefetch(buffer_size=AUTOTUNE) 
	
	return ds
```

**Comment by yongtang** [comment]
@gruossomonica Thanks for the report. I will take a look together with #745.

**Comment by yongtang** [comment]
@gruossomonica With PR #922 merged I think this issue is fixed. Can you give the nightly build `tensorflow-io-nightly` a try? Let me know if you encounter further issues.

**Comment by gruossomonica** [comment]
@yongtang thanks for the info. At the moment I can't try because I don't have access to the computer I used for my work (due to the pandemic). I'll try asap. I think we can close the issue and re-open it if the issue persists. Thanks again!
